### PR TITLE
[HandleError] rename msg field to action

### PR DESCRIPTION
**Pull request by savil**
this was a TODO and also briefly discussed in https://github.com/rust-bitcoin/rust-lightning/pull/43#pullrequestreview-135649551

I'm not fully sure how to remove the `Option`, and make it completely required. Would love suggestions. So, have omitted that for now. Plus, better to make smaller, incremental changes.

Test Plan:
`cargo build`
`cargo test`

### CODE REVIEW

File: src/ln/msgs.rs [hunk 1]
```diff
@@ -380,7 +380,7 @@ pub enum ErrorAction {
 
 pub struct HandleError { //TODO: rename me
 	pub err: &'static str,
-	pub msg: Option<ErrorAction>, //TODO: Make this required and rename it
+	pub action: Option<ErrorAction>, //TODO: Make this required and rename it
```
Comment (ariard) [inline src/ln/msgs.rs]: Update the TODO maybe :)

Comment (TheBlueMatt) [inline src/ln/msgs.rs]: Can remove the "rename it" TODO :)

**Comment by TheBlueMatt** [comment]
Note that "making it required" would be s/Option<ErrorAction>/ErrorAction/, but that will require filling in each action, which means figuring out exactly how we want to handle everything.

**Comment by ariard** [comment]
To remove the Option I think we need to fulfill incrementally all empty msg field of HandleError with actual errors or futures ones (like BanPeer or KeyError). I'll try to clean others TODO errors in `channelmanager` after #43

**Comment by savil** [comment]
- cool,  I updated the TODO. Let me know if anything else needs to be done.

- sorry for the commit thrash, I'm still figuring out the workflow for pull-requests on GitHub. I think I messed up git history for this branch, trying to resolve a merge-conflict from upstream. I updated my local `master` from upstream, and then did `git rebase master` to resolve the conflict. Needed to `git push -f origin <my branch>`, but surely there's a better way that avoids using `-f`. 

- to remove `Option`, agree that it'll need to be incremental. I'm so new to this, I don't feel I have enough context on each call site to figure out what kind of Error to insert there.

**Comment by ariard** [comment]
Have a look on Chap2,3,6 : http://labs.kernelconcepts.de/downloads/books/Pro%20Git%20-%20Scott%20Chacon.pdf ;)
You can squash your commits with git rebase -i HEAD~n (with n numbers of commits) to clean your git history.

Anyways don't hesitate to show up on IRC #rust-bitcoin, Freenode

**Comment by TheBlueMatt** [comment]
No, force-pushing to a feature branch is generally fine, its only really bad practice on a master branch (though I use my own personal fork's master branch as a feature branch, too, just don't force push to rust-bitcoin/rust-lightning).

File: src/ln/msgs.rs [hunk 2]
```diff
@@ -380,7 +380,7 @@ pub enum ErrorAction {
 
 pub struct HandleError { //TODO: rename me
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply src/ln/msgs.rs]: Hmm, I think eventually we need to separate out our error types, making this more of a MessageHandleError or PeerHandleError or something like that, and use a different type where that doesn't make sense.

**Pull request merged**